modules/ServerAPI/src/server/routes/ThingServiceRoutes.coffee

```diff
@@ -63,10 +65,55 @@ exports.setupRoutes = (app, loginRoutes) ->
 	app.post '/api/bulkPostThingsSaveFile', loginRoutes.ensureAuthenticated, exports.bulkPostThingsSaveFile
 	app.put '/api/bulkPutThingsSaveFile', loginRoutes.ensureAuthenticated, exports.bulkPutThingsSaveFile
 	app.delete '/api/things/:lsType/:lsKind/:idOrCodeName', loginRoutes.ensureAuthenticated, exports.deleteThing
+	app.get '/api/thingvalues/getThingValueById/:id', loginRoutes.ensureAuthenticated, exports.getThingValueById
+	app.get '/api/thingvalues/downloadThingBlobValueByID/:id', loginRoutes.ensureAuthenticated, exports.downloadThingBlobValueByID
+
 
 request = require 'request'
 config = require '../conf/compiled/conf.js'
 
+exports.getThingValueById = (req, resp) ->
+	exports.getlsValuesByIdInternal req.params.id, req.query, (statusCode, value) ->
+		resp.statusCode = statusCode
+		resp.json value
+		
+exports.getlsValuesByIdInternal = (id, params, callback) ->
+	serverUtilityFunctions = require './ServerUtilityFunctions.js'
+	baseurl = config.all.client.service.persistence.fullpath+"lsthingvalues/"+id
+	if params? && params.format?
+		baseurl += "?format=#{params.format}"
+	serverUtilityFunctions.getFromACASServerInternal baseurl, (statusCode, value) ->
+		callback(statusCode, value)
+
+exports.downloadThingBlobValueByID = (req, resp) ->
+	mime = require('mime');
+	fs = require('fs');
+	exports.getlsValuesInternal req.params.id, {format: "withblobvalue"}, (statusCode, value) ->
+		mimetype = mime.lookup(value.comments);
+		resp.setHeader('Content-disposition', 'attachment; filename=' + value.comments);
+		resp.setHeader('Content-type', mimetype);
+		buffer = new Buffer.from(Uint8Array.from(value.blobValue))
+		stream = exports.bufferToStream(buffer)
+		stream.on 'data', (chunk) ->
+			resp.send(chunk);
+		stream.on 'data', (chunk) ->
+			resp.status(200).send();
+
+exports.bufferToStream = (buffer) ->
+	Duplex = require('stream').Duplex
+	stream = new Duplex
+	stream.push buffer
+	stream.push null
+	stream
+
+exports.getlsValuesInternal = (id, params, callback) ->
```

Comment:
@brianbolt Is this a duplicate of `exports.getlsValuesByIdInternal` up above or is there something subtly different that I'm missing?

Reply:
This one is defining the function and the one above is calling.